src/scm/github.js

```diff
@@ -0,0 +1,427 @@
+    });
+  }
+
+  getGistCode() {
+    return new Promise((resolve, reject) => {
+      $.getJSON(
+        `${baseUrl}/gists/${context.branch}`,
+        { access_token: accessToken }
+      )
+      .then(resolve)
+      .fail(reject);
+    })
+    .then((response) => {
+      const promises = Object.keys(response.files).map((filename) => {
+        let file = response.files[filename];
+        return new Promise((resolve, reject) => {
+          if (file.truncated) {
+            $.getJSON(file.raw_url, {access_token: accessToken })
+            .then((content) => {
+              resolve({ file: filename, content: content});
+            })
+            .fail(reject)
+          } else {
+            resolve({file: filename, content: file.content});
+          }
+        });
+      });
+      return Promise.all(promises);
+    });
+  }
+
+  getRepos() {
+    return getAllItems(Promise.resolve({items: [], url: `${baseUrl}/user/repos?access_token=${accessToken}`}))
+    .then((response) => {
+      const repos = response.map((repo) => {
+        return { name : repo.name, fullName : repo.full_name }
+      });
+      //if current bind still existed, use it
+      const repo = context.bindRepo[context.id];
+      if (repo && $.inArray(repo.name, repos.map(repo => repo.name)) >= 0 ) {
+        context.repo = repo;
+      } else if (context.gist) {
+        context.repo = {
+          name: 'Using Gist',
+          fullName: 'gist',
+          gist: true
+        }
+      }
+      return repos;
+    })
+  }
+
+  createRepo() {
+    const repo = $('#new-repo-name').val();
+    const desc = $('#new-repo-desc').val();
+    const isPrivate = $('#new-repo-type').val() !== 'public';
+    const payload = {
+      name : repo,
+      description : desc,
+      auto_init : true,
+      private: isPrivate
+    }
+    if (!repo || repo === '') return;
+    new Promise((resolve, reject) => {
+      $.ajax({
+        url: `${baseUrl}/user/repos`,
+        headers: {
+          'Authorization': `token ${accessToken}`
+        },
+        method: 'POST',
+        crossDomain: true,
+        dataType: 'json',
+        contentType: 'application/json',
+        data: JSON.stringify(payload)
+      })
+      .then(resolve)
+      .fail(reject);
+    })
+    .then((response) => {
+      const repo = {
+        name : response.name,
+        fullName : response.full_name
+      };
+      context.repo = repo;
+      Object.assign(context.bindRepo, { [context.id] : repo });
+      if (context.bindBranch[context.id]) {
+        delete context.bindBranch[context.id];
+      }
+      chrome.storage.sync.set({ bindRepo: context.bindRepo });
+      return response;
+    })
+    .then(getGithubRepos)
+    .then(updateRepo)
+    .then(updateBranch)
+    .then(() => {
+      $('#new-repo-name').val('');
+      $('#new-repo-desc').val('');
+      $('#new-repo-type').val('public');
+      showAlert(`Successfully create new repository ${repo}`);
+    })
+    .catch((err) => {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/leonhartX/gas-github/blob/9b7bf2b597e67365f1e9002c784237b58657e4cc/src/scm/github.js#L315"></span>‘err’ is defined but never used.</td>
    </tr>
  </tbody>
</table>